### PR TITLE
remove some uneeded sleeps in tests

### PR DESCRIPTION
**Pull request by cpuguy83**

### CODE REVIEW

File: integration-cli/docker_api_logs_test.go [hunk 1]
```diff
@@ -5,26 +5,46 @@ import (
...
+	"strings"
+	"time"
 
 	"github.com/go-check/check"
 )
 
 func (s *DockerSuite) TestLogsApiWithStdout(c *check.C) {
-	name := "logs_test"
+	name := "log_test"
+	out, _ := dockerCmd(c, "run", "-d", "-t", "busybox", "/bin/sh", "-c", fmt.Sprintf("trap 'echo %s' SIGTERM; while true; do sleep 0.01; done", name))
```
Comment (LK4D4) [inline integration-cli/docker_api_logs_test.go]: Why not just `top` instead of loop?

Comment (cpuguy83) [inline integration-cli/docker_api_logs_test.go]: Because top was exiting with "143" when I sigterm'd

**Comment by jfrazelle** [comment]
ugh rebase sorry :(

**Comment by cpuguy83** [comment]
rebased

File: integration-cli/docker_api_logs_test.go [hunk 2]
```diff
@@ -5,24 +5,44 @@ import (
 	"fmt"
 	"net/http"
 	"os/exec"
+	"strings"
+	"time"
 
 	"github.com/go-check/check"
 )
 
 func (s *DockerSuite) TestLogsApiWithStdout(c *check.C) {
```
Comment (LK4D4) [inline integration-cli/docker_api_logs_test.go]: It is still took 11 seconds on jenkins. Do you have any idea why?

Comment (cpuguy83) [inline integration-cli/docker_api_logs_test.go]: Sure do... loop isn't exiting on SIGTERM :) Fix incoming.

Comment (cpuguy83) [inline integration-cli/docker_api_logs_test.go]: Updated, tests good. 0.61s now.

**Comment by LK4D4** [comment]
LGTM

File: integration-cli/docker_api_logs_test.go [hunk 3]
```diff
@@ -5,24 +5,44 @@ import (
...
 
-	status, body, err := sockRequest("GET", fmt.Sprintf("/containers/%s/logs?follow=1&stdout=1&timestamps=1", name), nil)
-	c.Assert(status, check.Equals, http.StatusOK)
-	c.Assert(err, check.IsNil)
+	go func() {
+		statusCode, body, err := sockRequest("GET", fmt.Sprintf("/containers/%s/logs?follow=1&stdout=1&timestamps=1", id), nil)
+		chLog <- logOut{body, statusCode, err}
+	}()
+
+	time.Sleep(100 * time.Millisecond)
```
Comment (duglin) [inline integration-cli/docker_api_logs_test.go]: You sure this isn't racey? What if it takes more than 100 milliseconds for the GET to get to the deamon?

Comment (cpuguy83) [inline integration-cli/docker_api_logs_test.go]: Technically yes it is inherently racey... do you see a good way around this?

Comment (duglin) [inline integration-cli/docker_api_logs_test.go]: Could we have the container print "name" to the screen, over and over. And then wait for it to show up in the logs?

Comment (cpuguy83) [inline integration-cli/docker_api_logs_test.go]: That's a pretty big change to this test... I'm not sure why it's doing what it's doing... let me do a git blame and see where it came in.

Comment (cpuguy83) [inline integration-cli/docker_api_logs_test.go]: Actually, this was moved from server_unit tests... I think we can make this change.

Comment (duglin) [inline integration-cli/docker_api_logs_test.go]: I _think_ its just checking to see that we see "name" in the logs and the sleep 10 was there to give the "docker logs" time to get started because if it started to see then we'd miss the one and only println. So having lots of println should be ok - I think.

File: integration-cli/docker_cli_wait_test.go [hunk 4]
```diff
@@ -44,19 +44,29 @@ func (s *DockerSuite) TestWaitNonBlockedExitZero(c *check.C) {
...
-	out, _, err = runCommandWithOutput(runCmd)
+	chWait := make(chan string)
+	go func() {
+		out, _, _ := runCommandWithOutput(exec.Command(dockerBinary, "wait", containerID))
+		chWait <- out
+	}()
 
-	if err != nil || strings.TrimSpace(out) != "0" {
-		c.Fatal("failed to set up container", out, err)
+	time.Sleep(100 * time.Millisecond)
```
Comment (duglin) [inline integration-cli/docker_cli_wait_test.go]: This one is racey too much I can't think of a good way to know when the 'wait' is there and blocking, so until someone else can think of a better way, this is ok.

File: integration-cli/docker_api_logs_test.go (hunk 3 above)
Comment (cpuguy83) [inline integration-cli/docker_api_logs_test.go]: Updated this one now that I saw the original intention of the test.  No more racey's here.

File: integration-cli/docker_cli_wait_test.go (hunk 4 above)
Comment (cpuguy83) [inline integration-cli/docker_cli_wait_test.go]: Yeah, and issue with it before was that if there was an issue with wait, it'd just hang and never timeout... which we could still do.

**Comment by duglin** [comment]
LGTM

**Comment by duglin** [comment]
so close - guess a rebase is needed

**Comment by cpuguy83** [comment]
FYI, this was merged but due to some weird UI sync bug with GH the PR itself got closed...and commits shown all jacked up after force-pushing to this brach (b/c the UI looked like there was a merge conflict, but there wasn't) after GH had had already merged the actual commit... and when this branch was force pushed GH automatically closed this PR.... weird stuff...

TLDR; this is merged.
